### PR TITLE
Add oVirt ovirt_hosts and ovirt_hosts_facts modules

### PR DESCRIPTION
**Pull request by machacekondra**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

- ovirt_hosts
- ovirt_hosts_facts
##### SUMMARY

This PR adds new modules for managing hosts in oVirt.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @machacekondra for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by machacekondra** [comment]
@mwperina @ryansb

File: cloud/ovirt/ovirt_hosts.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+description:
+    - "Module to manage hosts in oVirt"
+options:
+    name:
+        description:
+            - "Name of the the host to manage."
+        required: true
+    state:
+        description:
+            - "Should the host be present/absent/maintenance/upgraded"
```
Comment (mwperina) [inline cloud/ovirt/ovirt_hosts.py]: I'd rephrase to: 'State which should a host to be in after successful completion'

File: cloud/ovirt/ovirt_hosts.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+        default: present
+    comment:
+        description:
+            - "Description of the host."
+    cluster:
+        description:
+            - "Name of the cluster, where host should be created."
+    address:
+        description:
+            - "Host address. Can be IP address or FQDN."
```
Comment (mwperina) [inline cloud/ovirt/ovirt_hosts.py]: I'd rephrase to: 'Host address. It can be either FQDN (preferred) or IP address.'

File: cloud/ovirt/ovirt_hosts_facts.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        - "For example to search host X from datacenter Y use following pattern:
+           name=X and datacenter=Y"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all hosts named C<host**> from datacenter C(west):
```
Comment (mwperina) [inline cloud/ovirt/ovirt_hosts_facts.py]: I'd rephrase to: 'Gather facts about all hosts which names start with C<host> and belong to data center C(west):

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_hosts.py (hunk 1 above)
Comment (mwperina) [reply cloud/ovirt/ovirt_hosts.py]: Please rephrase as suggested

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_hosts.py (hunk 1 above)
Comment (machacekondra) [reply cloud/ovirt/ovirt_hosts.py]: done

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks!

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks again to @machacekondra. This module is going into community review.

We notice that you are one of the maintainers (@machacekondra) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]
